internal/hud/renderer.go

```diff
@@ -50,7 +50,13 @@ func formatPreciseDuration(d time.Duration) string {
 		return fmt.Sprintf("%dm", minutes)
 	}
 
-	return fmt.Sprintf("%ds", int(d.Seconds()))
+	seconds := int(d.Seconds())
+	if seconds > 10 {
+		return fmt.Sprintf("%ds", seconds)
+	}
+
+	fractionalSeconds := float64(d.Nanoseconds()) / (1000 * 1000 * 1000)
```

Comment:
maybe cleaner to do:
```suggestion
	fractionalSeconds := float64(d) / float64(time.Second)
```
? 🤷‍♀️